ddtrace/internal/telemetry/writer.py

```diff
@@ -116,8 +116,13 @@ def periodic(self):
                     exc_info=True,
                 )
 
-    def shutdown(self):
+    # PeriodicService.on_shutdown is a static method while TelemetryWriter.on_shutdown is not
```

Comment:
We do something similar in the runtime worker and the agent writer. Should `PeriodicService.on_shutdown` be static?

Reply:
I wonder if we need a test case for this regression?

e.g. verify that when a process shuts down we still get an app closing event sent?

Reply:
Done